module/functions/azure/aad/Assert-AzureAdSecurityGroup.Tests.ps1

```diff
@@ -0,0 +1,198 @@
+                EmailName = 'testgroup@nowhere.org'
+                Description = 'just a test group'
+                OwnersToAssignOnCreation = @()
+                StrictMode = $false
+            }
+            Assert-AzureAdSecurityGroup @testGroup
+
+            It "should not update the group" {
+                Assert-MockCalled _buildCreateRequest -Times 0
+                Assert-MockCalled _getGroupOwners -Times 0
+                Assert-MockCalled Invoke-AzRestMethod -Times 0
+            }
+        }
+
+        Context "Description updated (StrictMode=false)" {
+            $testGroup = @{
+                Name = 'testgroup'
+                EmailName = 'testgroup@nowhere.org'
+                Description = 'just a test group with a different description'
+                OwnersToAssignOnCreation = @()
+                StrictMode = $false
+            }
+            Assert-AzureAdSecurityGroup @testGroup
+
+            It "should not update the group" {
+                Assert-MockCalled _buildCreateRequest -Times 0
+                Assert-MockCalled _getGroupOwners -Times 0
+                Assert-MockCalled Invoke-AzRestMethod -Times 0
+            }
+        }
+
+        Context "Description updated (StrictMode=true)" {
+            $testGroup = @{
+                Name = 'testgroup'
+                EmailName = 'testgroup@nowhere.org'
+                Description = 'just a test group with a different description'
+                OwnersToAssignOnCreation = @()
+                StrictMode = $true
+            }
+            Assert-AzureAdSecurityGroup @testGroup
+
+            It "should update the group" {
+                Assert-MockCalled _buildCreateRequest -Times 0
+                Assert-MockCalled _getGroupOwners -Times 0
+                Assert-MockCalled Invoke-AzRestMethod -Times 1
+            }
+        }
+
+        Context "Default owners - No changes required" {
+            $testGroup = @{
+                Name = 'testgroup'
+                EmailName = 'testgroup@nowhere.org'
+                Description = 'just a test group'
+                OwnersToAssignOnCreation = @('11111111-1111-1111-1111-111111111111')
+                StrictMode = $false
+            }
+            Assert-AzureAdSecurityGroup @testGroup
+
+            It "should not update the group" {
+                Assert-MockCalled _buildCreateRequest -Times 0
+                Assert-MockCalled _getGroupOwners -Times 0
+                # Assert-MockCalled _buildUpdateRequest -Times 1
+                Assert-MockCalled Write-Warning -Times 0
+                Assert-MockCalled Invoke-AzRestMethod -Times 0
+            }
+        }
+
+        Context "Default owners - owner missing (StrictMode=false)" {
+            $testGroup = @{
+                Name = 'testgroup'
+                EmailName = 'testgroup@nowhere.org'
+                Description = 'just a test group'
+                OwnersToAssignOnCreation = @('22222222-2222-2222-2222-222222222222')
```

Comment:
This seems like a problematic concept in what otherwise feels more like a "desired state" kind of model. What's the use case for this (as opposed to just saying that certain users should be in the owners list)?

In any case, the test name feels like it might be wrong. I _think_ you'd only want it to ignore `OwnersToAssignOnCreation` if `StrictMode=false` AND the group already existed, right? There's a different expected outcome when `StrictMode=false` but we also had to create the group.

Reply:
I've done a pass on renaming the tests and also tweaked the implementation slightly to always log a warning when the required owners do not match those of an existing group.